android/301Project/app/src/main/java/com/example/a301project/ShoppingListController.java

```diff
@@ -35,6 +36,17 @@ public ShoppingListController() {
         mealplan_cr = db.collection("User").document(user.getEmail()).collection(mealplanCollectionName);
     }
 
+    /**
+     * Constructor for injecting a db for testing purposes
+     * @param db the database
+     */
+    public ShoppingListController(FirebaseFirestore db) {
+        this.db = db;
+        mealplan_cr = db.collection(collectionName);
```

Comment:
The variable name is mealplan_cr but collection name is set to ShoppingList

Reply:
yeah this doesn't make sense